### PR TITLE
Overall difficulty

### PR DESCRIPTION
**Pull request by Shawdooow**

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/OsuHitObject.cs [hunk 1]
```diff
@@ -78,6 +78,10 @@ protected override void ApplyDefaultsToSelf(ControlPointInfo controlPointInfo, B
             TimePreempt = (float)BeatmapDifficulty.DifficultyRange(difficulty.ApproachRate, 1800, 1200, 450);
             TimeFadein = (float)BeatmapDifficulty.DifficultyRange(difficulty.ApproachRate, 1200, 800, 300);
 
+            HitWindow50 = BeatmapDifficulty.DifficultyRange(difficulty.OverallDifficulty, 99.5, 149.5, 199.5);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/OsuHitObject.cs]: would be good to explain the 0.5 offset. not even i'm sure why it's here.

Comment (Shawdooow) [reply osu.Game.Rulesets.Osu/Objects/OsuHitObject.cs]: the chart on the wiki (<https://osu.ppy.sh/help/wiki/Beatmap_Editor/Song_Setup>) has them listed with the 0.5 which is where i got these all from

Comment (peppy) [reply osu.Game.Rulesets.Osu/Objects/OsuHitObject.cs]: correct values from stable:

```csharp
            HitWindow50 = (int)MapDifficultyRange(Beatmap.DifficultyOverall, 200, 150, 100);
            HitWindow100 = (int)MapDifficultyRange(Beatmap.DifficultyOverall, 140, 100, 60);
            HitWindow300 = (int)MapDifficultyRange(Beatmap.DifficultyOverall, 80, 50, 20);
```

**Review by smoogipoo — APPROVED** [review]
We're going to need to clean up that InitialCircle thing hey :P

**Pull request merged**